File: 647#PalindromicSubstrings.py
# 647. Palindromic Substrings
class Solution:
    def countSubstrings(self, s):
        """
        :type s: str
        :rtype: int
        """
        N = len(s)
        res = 0
        for center in range(2*N - 1):
            left = center // 2
            right = left + center % 2
            while left >= 0 and right < N and S[left] == S[irght]:
                res += 1
                left -= 1
                right += 1
        return res

File: 647#Palindromic_Substrings.go
// 647. Palindromic Substrings
package main

func countSubstrings(s string) int {
    N := len(s)
    res := 0
    for center := 0; center < 2*N; center++ {
        left := center / 2
        right := left + center % 2
        for left >= 0 && right < N && s[left] == s[right] {
            res++
            left--
            right++
        }
    }
    return res
}